pkgs/build-support/rust/replace-workspace-values.py

```diff
@@ -0,0 +1,57 @@
+import sys
+import toml
+
+crate_manifest = toml.load(sys.argv[1])
+workspace_manifest = toml.load(sys.argv[2])["workspace"]
+
+
+def replace_key(table, section, key):
+    if "workspace" in table[key] and table[key]["workspace"] is True:
+        print("replacing " + key)
+
+        replaced = table[key]
+        del replaced["workspace"]
+
+        workspace_copy = workspace_manifest[section][key]
+
+        if section == "dependencies":
+            crate_features = replaced.get("features")
+
+            if type(workspace_copy) is str:
+                replaced["version"] = workspace_copy
+            else:
+                replaced.update(workspace_copy)
+
+            merged_features = []
+
+            if crate_features is not None:
+                merged_features.extend(crate_features)
+
+            if (workspace_features := workspace_copy.get("features")) is not None:
+                merged_features.extend(workspace_features)
```

Comment:
I think this should do the same thing, and we can move this logic (and whatever `dict.fromkeys` does I'm assuming) after (or before if we mutate workspace_copy) `replaced.update(workspace_copy)`
```suggestion
            merged_features = set((crate_features or []) + (workspace_copy.get("features") or []))
```

Reply:
You know, when writing that, I was thinking to myself "if only Python had `or`." 😄